src/conn/conn_compact.c

```diff
@@ -21,21 +22,256 @@ __compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_list_insert --
+ *     Insert compaction statistics for a file into the background compact list.
+ */
+static void
+__background_compact_list_insert(WT_SESSION_IMPL *session, WT_BACKGROUND_COMPACT_STAT *compact_stat)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(compact_stat->uri, strlen(compact_stat->uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    TAILQ_INSERT_HEAD(&(conn)->background_compact.compactqh, compact_stat, q);
+    TAILQ_INSERT_HEAD(&(conn)->background_compact.compacthash[bucket], compact_stat, hashq);
+    ++(conn)->background_compact.file_count;
+}
+
+/*
+ * __background_compact_list_remove --
+ *     Remove compaction statistics for a file from the background compact list.
+ */
+static void
+__background_compact_list_remove(WT_SESSION_IMPL *session, WT_BACKGROUND_COMPACT_STAT *compact_stat)
+{
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(compact_stat->uri, strlen(compact_stat->uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    TAILQ_REMOVE(&(conn)->background_compact.compactqh, compact_stat, q);
+    TAILQ_REMOVE(&(conn)->background_compact.compacthash[bucket], compact_stat, hashq);
+    WT_ASSERT(session, (conn)->background_compact.file_count > 0);
+    --(conn)->background_compact.file_count;
+
+    __wt_free(session, compact_stat->uri);
+    __wt_free(session, compact_stat);
+}
+
+/*
+ * __background_compact_get_stat --
+ *     Get the statistics for the given uri.
+ */
+static WT_BACKGROUND_COMPACT_STAT *
+__background_compact_get_stat(WT_SESSION_IMPL *session, const char *uri, int64_t id)
+{
+    WT_BACKGROUND_COMPACT_STAT *compact_stat, *temp_compact_stat;
+    WT_CONNECTION_IMPL *conn;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    hash = __wt_hash_city64(uri, strlen(uri));
+    bucket = hash & (conn->hash_size - 1);
+
+    /* Find the uri in the files compacted list. */
+    TAILQ_FOREACH_SAFE(
+      compact_stat, &conn->background_compact.compacthash[bucket], hashq, temp_compact_stat)
+    if (strcmp(uri, compact_stat->uri) == 0) {
```

Comment:
The indenting doesn't look right here. This `if` is within the purview of the above `FOREACH`.

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9541/commits/112ccd9c97af7114df00e6b89c772d59b5d29444.

Reply:
I think the auto-formatting is undoing indentation. You might need braces, since there is an unfamiliar loop construct.

Reply:
@Sean04, I would like to make sure and verify the indentation here. Do we need a `{` after `TAILQ_FOREACH_SAFE`? I can see an example in the `block_cache.c` file where there is the following code:
```
            TAILQ_FOREACH_SAFE(blkcache_item, &blkcache->hash[i], hashq, blkcache_item_tmp)
            {
...
            }
```